File: R/scatterplots.r
data <- read.csv(file="../data/commodityprices.csv",head=TRUE,sep=",")
colnames(data)  <- c("Year","Week","Butter","Cheese","Potatoes (maincrop)","Potatoes (earlies)")#Fix for text importing
# Remove empty time points for Butter
data <- data[!is.na(data$Butter),]

## Load library
library(ggplot2)

## Scatter Plot - Butter: Week vs. Price (coloured by year)
# Initialise plot
ggplot(data=data) +
# Plot points using x and y with Year as colour
geom_point(aes(x=Week,y=Butter,color=Year)) +
# Legend is added automatically
# Set x limits
expand_limits(x = c(1,52)) +
# Add axis labels
xlab("Week Number") +
ylab("Cost of Butter, £/tonnes")

## Line Plot - Butter: Week vs. Price (one line per year)
# Initialise plot
ggplot() + 
# Plot line using x and y with Year as colour
geom_line(data=data[data$Year==2013,], aes(x=Week,y=Butter,color=Year)) +
geom_line(data=data[data$Year==2014,], aes(Week,Butter,color=Year)) +
geom_line(data=data[data$Year==2015,], aes(Week,Butter,color=Year)) +
# Specify line style for 2016
geom_line(data=data[data$Year==2016,], aes(Week,Butter,color=Year), linetype="dashed") +
# Legend is added automatically
# Set x limits
expand_limits(x = c(1,52)) +
# Add axis labels
xlab("Week Number") +
ylab("Cost of Butter, £/tonnes")

## Line Plot w/ Error Bars - Butter: Week vs. Price (one line per year)
# Note that we've used a random error here created with rand for illustration purposes, for different values at each point use a 
# Just 2013 data for this example
data13 = data[data$Year==2013,]
# Generate Random Errors
data13$se <- sample (1:100, nrow(data13), replace=TRUE)
#Initialise plot
ggplot() + 
# Plot line using x and y with Year as colour
geom_line(data=data13, aes(x=Week,y=Butter,color=Year)) +
# Add error bars
geom_errorbar(data=data13, aes(x=Week,y=Butter,ymin=Butter-se, ymax=Butter+se), width=0.5) +
# Legend is added automatically
# Set x limits
expand_limits(x = c(1,52)) +
# Add axis labels
xlab("Week Number") +
ylab("Cost of Butter, £/tonnes")

File: MATLAB/scatterplots.m
%% Load commodities data
filename = '../data/commodityprices.csv';
data = csvread(filename,1);
headings = {'Year','Week','Butter','Cheese','Potatoes (main)', 'Potatoes (early)'};

%% Scatter Plot - Butter: Week vs. Price (coloured by year)
figure
% Remove empty time points
data(data(:,3)==0,:) = [];
scatter(data(:,2),data(:,3),20,data(:,1),'o','filled');
% Change colours
colormap('jet')
% Set x-axis limits and add label
xlim([1,52])
xlabel('Week Number')
% Set y-axis label
ylabel('Cost of Butter, £/tonnes')

%% Line Plot - Butter: Week vs. Price (one line per year)
figure
% Remove empty time points
data(data(:,3)==0,:) = [];
for y=2013:2016
    ydata = data(data(:,1)==y,:);
    % Specify line style for 2016
    if y==2016
        plot(ydata(:,2),ydata(:,3),'--');
    else
        plot(ydata(:,2),ydata(:,3));
    end
    hold on;
end
% Add legend
legend('2013','2014','2015','2016')
% Set x-axis limits and add label
xlim([1,52])
xlabel('Week Number')
% Set y-axis label
ylabel('Cost of Butter, £/tonnes')

%% Line Plot w/ Error Bars - Butter: Week vs. Price (one line per year)
% Note that we've used a random error here created with rand for
% illustration purposes, for different values at each point use a
figure
% Remove empty time points
data(data(:,3)==0,:) = [];
for y=2013:2016
    ydata = data(data(:,1)==y,:);
    % Specify line style for 2016
    if y==2016
        plot(ydata(:,2),ydata(:,3),'--');
    else
        plot(ydata(:,2),ydata(:,3));
    end
    errorbar(ydata(:,2),ydata(:,3),50*rand(1,length(ydata(:,2))));
    hold on;
end
set(gca,'FontSize',16);%Set a font size for axis tick labels, note gca means 'get current axis;
% Add legend
legend({'2013','2014','2015','2016'},'Location','bestoutside','Orientation','horizontal','FontSize',16)
% Set x-axis limits and add label
xlim([1,52])
xlabel('Week Number','FontSize',16)
% Set y-axis label
ylabel('Cost of Butter, £/tonnes','FontSize',16)
